contrib/llvm-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,98 @@
+if (APPLE OR NOT ARCH_AMD64 OR SANITIZE STREQUAL "undefined")
+    set (ENABLE_EMBEDDED_COMPILER_DEFAULT OFF)
+else()
+    set (ENABLE_EMBEDDED_COMPILER_DEFAULT ON)
+endif()
+option (ENABLE_EMBEDDED_COMPILER "Enable support for 'compile_expressions' option for query execution" ${ENABLE_EMBEDDED_COMPILER_DEFAULT})
+
+if (NOT ENABLE_EMBEDDED_COMPILER)
+    set (USE_EMBEDDED_COMPILER 0)
+    return()
+endif()
+
+set (LLVM_FOUND 1)
+set (LLVM_VERSION "12.0.0bundled")
+set (LLVM_INCLUDE_DIRS
+    "${ClickHouse_SOURCE_DIR}/contrib/llvm/llvm/include"
+    "${ClickHouse_BINARY_DIR}/contrib/llvm/llvm/include"
+)
+set (LLVM_LIBRARY_DIRS "${ClickHouse_BINARY_DIR}/contrib/llvm/llvm")
+
+# This list was generated by listing all LLVM libraries, compiling the binary and removing all libraries while it still compiles.
+set (REQUIRED_LLVM_LIBRARIES
+    LLVMExecutionEngine
+    LLVMRuntimeDyld
+    LLVMX86CodeGen
+    LLVMX86Desc
+    LLVMX86Info
+    LLVMAsmPrinter
+    LLVMDebugInfoDWARF
+    LLVMGlobalISel
+    LLVMSelectionDAG
+    LLVMMCDisassembler
+    LLVMPasses
+    LLVMCodeGen
+    LLVMipo
+    LLVMBitWriter
+    LLVMInstrumentation
+    LLVMScalarOpts
+    LLVMAggressiveInstCombine
+    LLVMInstCombine
+    LLVMVectorize
+    LLVMTransformUtils
+    LLVMTarget
+    LLVMAnalysis
+    LLVMProfileData
+    LLVMObject
+    LLVMBitReader
+    LLVMCore
+    LLVMRemarks
+    LLVMBitstreamReader
+    LLVMMCParser
+    LLVMMC
+    LLVMBinaryFormat
+    LLVMDebugInfoCodeView
+    LLVMSupport
+    LLVMDemangle
+)
+
+#function(llvm_libs_all REQUIRED_LLVM_LIBRARIES)
+#    llvm_map_components_to_libnames (result all)
+#    if (USE_STATIC_LIBRARIES OR NOT "LLVM" IN_LIST result)
+#        list (REMOVE_ITEM result "LTO" "LLVM")
+#    else()
+#        set (result "LLVM")
+#    endif ()
+#    list (APPEND result ${CMAKE_DL_LIBS} ch_contrib::zlib)
+#    set (${REQUIRED_LLVM_LIBRARIES} ${result} PARENT_SCOPE)
+#endfunction()
+
+message (STATUS "LLVM include Directory: ${LLVM_INCLUDE_DIRS}")
+message (STATUS "LLVM library Directory: ${LLVM_LIBRARY_DIRS}")
+message (STATUS "LLVM C++ compiler flags: ${LLVM_CXXFLAGS}")
+
+# ld: unknown option: --color-diagnostics
+if (APPLE)
+    set (LINKER_SUPPORTS_COLOR_DIAGNOSTICS 0 CACHE INTERNAL "")
+endif ()
+
+# Do not adjust RPATH in llvm, since then it will not be able to find libcxx/libcxxabi/libunwind
+set (CMAKE_INSTALL_RPATH "ON")
+set (LLVM_COMPILER_CHECKED 1 CACHE INTERNAL "")
+set (LLVM_ENABLE_EH 1 CACHE INTERNAL "")
+set (LLVM_ENABLE_RTTI 1 CACHE INTERNAL "")
+set (LLVM_ENABLE_PIC 0 CACHE INTERNAL "")
+set (LLVM_TARGETS_TO_BUILD "X86;AArch64" CACHE STRING "")
```

Comment:
Why you build both x86 and aarch64? As I understand we have information about target and host architecture.